nixos/modules/programs/zsh/zsh.nix

```diff
@@ -69,7 +69,7 @@ in
 
       promptInit = mkOption {
         default = ''
-          autoload -U promptinit && promptinit && prompt walters
+          autoload -U promptinit && promptinit
```

Comment:
I know other distros do not that anything, but they also do not have configuration management that let user's override.
The default prompt of zsh is almost useless. Could we at least have something that is consistent with our bash prompt?

```
          # Provide a nice prompt if the terminal supports it.
          if [ "$TERM" != "dumb" -o -n "$INSIDE_EMACS" ]; then
            PROMPT_COLOR="1;31m"
            let $UID && PROMPT_COLOR="1;32m"
            PS1="\n\[\033[$PROMPT_COLOR\][\u@\h:\w]\\$\[\033[0m\] "
            if test "$TERM" = "xterm"; then
              PS1="\[\033]2;\h:\u:\w\007\]$PS1"
            fi
          fi
```

it does not have to be fancy pants, but it should be a bit saner to what zsh provide by default.
The reason is, that in automatically configured systems, you don't want to spend time on setting up dotfiles.
Also when users configure there shells the first time they also should not suffer from eye bleeding.

Reply:
I'm not enough of a zsh expert to know how to write an idiomatic configuration, but I would like to avoid having a prompt theme injected by default since this changes the effect of dotfiles atop a regular install compared to other distros.

To me, it seems like automated deploys using zsh will also have custom configuration for it, since zsh is not a default shell in the first place (opinionated sysadmins tend to come with their own dotfiles :P).

Unless someone knows of a way to make the default zsh configuration friendly which will *not* interfere with other configuration-provided zsh nor user dotfiles, I think we should merge this as-is. Users are used to adding their own config, but shouldn't have to contend with unexpected pre-existing configuration that isn't familiar from other distros or zsh experience.

Reply:
Anything that runs afterwards, including traditional dotfiles like `~/.zshrc` will override the system defaults. Alternatively it can be disabled globally with `programs.zsh.promptInit = mkForce "";`, the option should probably be `types.str` instead of `types.lines` then regular assignment would just work.

Reply:
Note that disabling zsh on global system level and leave it enabled only at home manager user level, eliminate this weird (IMHO very ucommon) option.